library-model/library-model-generator/src/main/java/com/uber/nullaway/libmodel/LibraryModelGenerator.java

```diff
@@ -197,6 +201,17 @@ public void visit(ClassOrInterfaceDeclaration cid, Void arg) {
                   this.isNullMarked = true;
                 }
               });
+      ImmutableMap<Integer, ImmutableSet<String>> genericParamAnnotationsMap =
+          getGenericTypeParameterNullableUpperBounds(cid);
+      /*
+       We insert a specialized MethodAnnotationsRecord object to store the generic type parameter nullability
+       information for the class by using an identifier that can never be an actual method name.
+      */
+      if (this.isNullMarked && !genericParamAnnotationsMap.isEmpty()) {
```

Comment:
Should we be checking if the class is null marked before getting the upper bounds at line 204, for efficiency?